shotover-proxy/src/transforms/query_counter.rs

```diff
@@ -46,24 +54,45 @@ impl Transform for QueryCounter {
                         };
                         counter!("query_count", 1, "name" => self.counter_name.clone(), "query" => query_type, "type" => "cassandra");
                     }
-                    ASTHolder::Commands(List(commands)) => {
-                        if let Some(v) = commands.get(0) {
-                            let command = format!("{v:?}");
-                            counter!("query_count", 1, "name" => self.counter_name.clone(), "query" => command.to_ascii_uppercase(), "type" => "redis");
-                        } else {
-                            counter!("query_count", 1, "name" => self.counter_name.clone(), "query" => "empty", "type" => "redis");
-                        }
-                    }
-                    _ => {
-                        counter!("query_count", 1, "name" => self.counter_name.clone(), "query" => "unknown", "type" => "unknown");
+                }
+                Frame::Cassandra(_) => {
+                    counter!("query_count", 1, "name" => self.counter_name.clone(), "query" => "unknown", "type" => "cassandra")
+                }
+                Frame::Redis(frame) => {
+                    if let Some(query_type) = get_redis_query_type(frame) {
+                        counter!("query_count", 1, "name" => self.counter_name.clone(), "query" => query_type, "type" => "redis");
+                    } else {
+                        counter!("query_count", 1, "name" => self.counter_name.clone(), "query" => "unknown", "type" => "redis");
                     }
                 }
+                Frame::None => {
+                    counter!("query_count", 1, "name" => self.counter_name.clone(), "query" => "unknown", "type" => "none")
+                }
             }
         }
 
         message_wrapper.call_next_transform().await
     }
 }
+fn get_redis_query_type(frame: &RedisFrame) -> Option<String> {
```

Comment:
Should this be using something similar to `Message::get_query_type` and returning an enum of `QueryType` instead? That way developers using the metrics will have all query types in the same format and not have to handle different ones for Cassandra and Redis. e.g. 

Redis `"SET"` & Cassandra `"INSERT"` or `"UPDATE"` -> `QueryType::Write`
Redis `"GET"` & Cassandra `"SELECT"` -> `QueryType::Read`

That said, won't block this PR on it as it's just progress towards the `MessageDetails` rewrite and didn't change what was already implemented.